### ISSUE TITLE
Integrate with Unity ML Agents

### ISSUE THREAD

**Issue by gavinuhma** [OpenMined/OpenMined#191]
Unity has a new project for training intelligent agents through a python API: https://github.com/Unity-Technologies/ml-agents

Wouldn't is be cool if Unity's ML project could take advantage of their own amazing GPU portability?

We're not that far off:

- [They keep an open socket between python<>unity, like we do.](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Unity-Agents---Python-API.md)
- [They are currently using tensorflow, but we could swap that out for OpenMined.](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Unity-Agents-Overview.md)

The integration would give us tons of awesome game environments to play with: https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Example-Environments.md

Imagine the fun hackathons we would have?! :) 

Supporting reinforcement learning is our last major blocker before we could tackle this.

**Comment by iamtrask** [comment]
Oh man this would be super cool!!! The other piece we're missing is REINFORCE in our autograd system - i mean technically you could sortof do it by hand - but it'd be a lot nicer if we had it implemented automagically. I believe PyTorch just upgraded their API for this and DyNet has some nice conventions we can take inspiration from as well.

**Comment by iamtrask** [comment]
Reference on Reinforcement Learning: karpathy.github.io/2016/05/31/rl/

**Comment by diegoalejogm** [comment]
What's the REINFORCE functionality in this case? Policy gradient?

**Comment by awjuliani** [comment]
Hi All! I am working at Unity on the ML-Agent's project. We'd love to have OpenMined as an inference (and eventually training) backend for ML-Agents games on Unity. If anyone starts to take this up, feel free to reach out to me about the Unity/Agents side of things.

**Comment by iamtrask** [comment]
@diegoalejogm we're initially starting with Policy Gradient for training - working on a prototype now. I'll let you know when i have a work-in-progress branch
